### ISSUE TITLE
It's possible to collide with floors/ceilings behind a portal

### ISSUE THREAD

**Issue by Ninja-Cookie** [momentum-mod/game#1611]
### Describe the bug

Similar to bugging on a flat wall normally in TF2, you can collide with a floor or ceiling by hold against a wall + hold a strafe direction while trying to go through a portal to - for some reason - collide with the floor under the portal you're about to enter, when usually you would not.

Even at 4 units away from the floor you do not collide with the floor under the portal normally.

For general cases you should have the portal well above a floor anyway, however it's still a strange collision bug.

https://user-images.githubusercontent.com/62808028/145690936-7fc4fe55-0027-4856-80e3-23fd3cf2fa38.mp4

https://user-images.githubusercontent.com/62808028/145689904-7ec3a4a0-6098-455e-a284-b630ecbf5033.mp4

![image](https://user-images.githubusercontent.com/62808028/145691131-985d9e00-af83-44aa-bf16-14e0bdb6dc8f.png)

### How To Reproduce

1. Have a portal placed in a position where you can fall against a wall, this only needs to be at the portals entrance side.
2. Have the portal be above a floor within at least 64 units (happens more consistently the closer to the floor the portal is)
3. Fall against a wall as you enter a portal, and hold W+A/D (Forward+Left/Right)

### Expected Behavior

Go through the portal normally as you would if not against a wall.

### Operating System

Windows

### Renderer

DX11

**Comment by jason-e** [comment]
@Ninja-Cookie There were some general collision fixes made recently -- do you know if this bug still occurs? If so, could you link an example map?

**Comment by Ninja-Cookie** [comment]
> @Ninja-Cookie There were some general collision fixes made recently -- do you know if this bug still occurs? If so, could you link an example map?

Yup it still happens, load this map and just hold W, or do mom_restart and hold W

[test_portals.zip](https://github.com/momentum-mod/game/files/8792215/test_portals.zip)

**Comment by hexaflexahexagon** [comment]
wonder if this is related: https://github.com/ChaosInitiative/Portal-2-Community-Edition/issues/934
